src/main/java/net/minecraftforge/client/event/sound/BackgroundMusicSelectionEvent.java

```diff
@@ -0,0 +1,94 @@
+package net.minecraftforge.client.event.sound;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.audio.BackgroundMusicSelector;
+import net.minecraft.client.audio.ISound;
+import net.minecraft.client.audio.MusicTicker;
+import net.minecraft.client.audio.SimpleSound;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * Fired whenever {@link MusicTicker} tries to get the current
+ * {@link BackgroundMusicSelector} for the current situation.
+ * 
+ * When the setter below is called, the current background music selector
+ * is replaced with the user set one. The selector will wait until the current
+ * music is finished playing followed by a random delay specified by the new
+ * selector or if {@link BackgroundMusicSelector#shouldReplaceCurrentMusic}
+ * on the new selector is true.
+ * 
+ * If the setter is never called, the background music selector will default to
+ * the client call from {@link Minecraft#getBackgroundMusicSelector}.
+ * */
+public class BackgroundMusicSelectionEvent extends Event
+{
+    private final MusicTicker musicTicker;
+    private final Minecraft mc;
+    @Nullable
+    private final ISound currentMusic;
+    private final int timeUntilNextMusic;
```

Comment:
Why is this needed? Which music should be played does not depend on this field at all.